xds/src/main/java/io/grpc/xds/orca/OrcaMetricReportingServerInterceptor.java

```diff
@@ -77,6 +85,10 @@ public <ReqT, RespT> Listener<ReqT> interceptCall(
           public void close(Status status, Metadata trailers) {
             OrcaLoadReport report = fromInternalReport(
                 InternalCallMetricRecorder.finalizeAndDump2(finalCallMetricRecorder));
+            if (metricRecorder != null) {
```

Comment:
metricRecorder was just for out of band, i'm surprised it gets mixed. Anything that you can't do with CallMetricRecorder itself?

Reply:
According to go/grpc-backend-metrics-update#heading=h.mmgva5nkuphg, The new ServerMetricRecorder API in the docs is `MetricRecorder` in Java,  which could be mixed and used for both OOB and per-call. 

From go/grpc-backend-metrics-update#heading=h.jlcmd1ovwjb, if `MetricRecorder` is set, then we merge metrics such that metrics from `CallMetricRecorder` takes higher precedence over `MetricRecorder`

Reply:
Thanks. The doc sounds fine.
First, although this is not a global interceptor, nor did we have the contract that it is global. But conceptually I guess it is fine to assume it is the only one and make `MetricRecorder` is static. To me, it is better to make `MetricRecorder` non-static but I can't argue static has potential problem.

For thread safety, the interceptor can be in call executor, while `setMetricRecorder` can be in another thread. So it is necessary to make it volatile.  Correct me if I am wrong. @ejona86

Reply:
Updated `MetricRecorder` field to be non-static and made it `volatile` for thread safety and updated corresponding unit tests.

Reply:
See my comment about removing the static. We'll need to deal with this in the design.

Reply:
In latest commit, updated `setMetricRecorder`  be non-static and made the field `metricRecorder` non-static and volatile for thread safety and updated corresponding unit tests.